File: talentflow-api/app/schemas/job.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import date, datetime


class JobResponse(BaseModel):
    id: str
    slug: str
    title: str
    description: str
    location: Optional[str] = None
    employment_type: Optional[str] = None
    work_model: Optional[str] = None
    responsibilities: Optional[str] = None
    requirements: Optional[str] = None
    benefits: Optional[str] = None
    application_email: Optional[str] = None
    application_subject: Optional[str] = None
    deadline: Optional[str] = None
    required_skills: Optional[str] = None
    is_active: bool
    created_at: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class PublicJobResponse(BaseModel):
    id: str
    slug: str
    title: str
    description: str
    location: Optional[str] = None
    employment_type: Optional[str] = None
    work_model: Optional[str] = None
    responsibilities: Optional[str] = None
    requirements: Optional[str] = None
    benefits: Optional[str] = None
    deadline: Optional[str] = None
    required_skills: Optional[str] = None
    created_at: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: talentflow-web/src/types/job.ts
export interface Job {
  id: string;
  slug?: string;
  title: string;
  description: string;
  location?: string | null;
  employment_type?: string | null;
  work_model?: string | null;
  responsibilities?: string | null;
  requirements?: string | null;
  benefits?: string | null;
  application_email?: string | null;
  application_subject?: string | null;
  deadline?: string | null;
  required_skills?: string | null;
  is_active: boolean;
  created_at?: string | null;
}

export interface JobFormData {
  title: string;
  description: string;
  location?: string;
  employment_type?: string;
  work_model?: string;
  responsibilities?: string;
  requirements?: string;
  benefits?: string;
  application_email?: string;
  application_subject?: string;
  deadline?: string;
  required_skills?: string;
}
